driver-core/src/main/com/mongodb/internal/operation/BaseFindAndModifyOperation.java

```diff
@@ -122,22 +140,231 @@ public boolean isRetryWrites() {
+     *
+     * @return the collation options
+     * @since 3.4
+     * @mongodb.server.release 3.4
+     */
+    public Collation getCollation() {
+        return collation;
+    }
+
+    /**
+     * Returns the hint for which index to use. The default is not to set a hint.
+     *
+     * @return the hint
+     * @since 4.1
+     */
+    @Nullable
+    public Bson getHint() {
+        return hint;
+    }
+
+    /**
+     * Sets the hint for which index to use. A null value means no hint is set.
+     *
+     * @param hint the hint
+     * @return this
+     * @since 4.1
+     */
+    public BaseFindAndModifyOperation<T> hint(@Nullable final Bson hint) {
+        this.hint = hint;
+        return this;
+    }
+
+    /**
+     * Gets the hint string to apply.
+     *
+     * @return the hint string, which should be the name of an existing index
+     * @since 4.1
+     */
+    @Nullable
+    public String getHintString() {
+        return hintString;
+    }
+
+    /**
+     * Sets the hint to apply.
+     *
+     * @param hint the name of the index which should be used for the operation
+     * @return this
+     * @since 4.1
+     */
+    public BaseFindAndModifyOperation<T> hintString(@Nullable final String hint) {
+        this.hintString = hint;
+        return this;
+    }
+
+    /**
+     * Sets the collation options
+     *
+     * <p>A null value represents the server default.</p>
+     * @param collation the collation options to use
+     * @return this
+     * @since 3.4
+     * @mongodb.server.release 3.4
+     */
+    public BaseFindAndModifyOperation<T> collation(final Collation collation) {
+        this.collation = collation;
+        return this;
+    }
+
+    /**
+     * @return comment for this operation. A null value means no comment is set.
+     * @since 4.6
+     * @mongodb.server.release 4.4
+     */
+    public BsonValue getComment() {
+        return comment;
     }
 
-    protected void addWriteConcernToCommand(final ConnectionDescription connectionDescription, final BsonDocument commandDocument,
-                                            final SessionContext sessionContext) {
-        if (getWriteConcern().isAcknowledged() && !getWriteConcern().isServerDefault()
-                && serverIsAtLeastVersionThreeDotTwo(connectionDescription) && !sessionContext.hasActiveTransaction()) {
-            commandDocument.put("writeConcern", getWriteConcern().asDocument());
-        }
+    /**
+     * Sets the comment for this operation. A null value means no comment is set.
+     *
+     * @param comment the comment
+     * @return this
+     * @since 4.6
+     * @mongodb.server.release 4.4
+     */
+    public BaseFindAndModifyOperation<T> comment(final BsonValue comment) {
+        this.comment = comment;
+        return this;
     }
 
     protected abstract FieldNameValidator getFieldNameValidator();
+
+    protected abstract void specializeCommand(BsonDocument initialCommand, ConnectionDescription connectionDescription);
```

Comment:
Nit: since this is an internal API, it's safe to remove the currently-unused `connectionDescription` parameter.

Reply:
You could also make it not abstract since 2 of 3 subclasses don't require it.

Reply:
Never mind.  I mis-read the diffs and thought that two of the three implementations were empty methods.